x-pack/plugins/telemetry_collection_xpack/schema/xpack_plugins.json

```diff
@@ -1644,6 +1644,55 @@
             }
           }
         },
+        "count_rules_by_execution_status": {
+          "properties": {
+            "success": {
+              "type": "long"
+            },
+            "error": {
+              "type": "long"
+            },
+            "warning": {
+              "type": "long"
+            }
+          }
+        },
+        "count_rules_with_tags": {
+          "type": "long"
+        },
+        "count_rules_by_notify_when": {
+          "properties": {
+            "on_action_group_change": {
+              "type": "long"
+            },
+            "on_active_alert": {
+              "type": "long"
+            },
+            "on_throttle_interval": {
+              "type": "long"
+            }
+          }
+        },
+        "count_rules_snoozed": {
+          "type": "long"
+        },
+        "count_rules_muted": {
+          "type": "long"
+        },
+        "count_rules_with_muted_alerts": {
+          "type": "long"
+        },
+        "count_connector_types_by_consumers": {
+          "properties": {
+            "DYNAMIC_KEY": {
```

Comment:
This is going to look for a field `DYNAMIC_KEY` and apply the mapping to that, so the mapping will not be applied to data like `siem.__email` or `alerts.__email`. With other items that we break down by rule type id, we have to explicitly map each possible value.

However, since we plan to copy these to the custom response ops index via the indexer, maybe having this mapping doesn't matter as much anymore? WDYT @mikecote

Reply:
From what I recall, the introduction of `DYNAMIC_KEY` was to allow telemetry to be indexed without failures (using strict mappings?). I wonder if this has changed now where they just store the data within the _source unmapped.

@elastic/kibana-core can you weigh in on whether or not we still need to use `DYNAMIC_KEY` to make such data accessible in our custom telemetry indexer?

Reply:
cc @afharo @Bamieh

Reply:
Thank you for the ping! I think there is some confusion between the schema and the mappings:

Kibana describes the _schema_ as a way to declare a contract of the format of the document we ship. This may or may not result in a direct mapping approach.

-----

Re `DYNAMIC_KEY`, it was introduced to allow Kibana developers to specify tree-shaped structures where the keys are dynamic/arbitrarily built. Whether we use `flattened` type, use dynamic templates or add the known keys to the mappings, it's up to the analysis end to decide. The important thing is that, from the Kibana POV, it's sending an object with an arbitrary set of keys.

-----

Re the keys including a dot inside, that's a good point, @ymao1. We allow shipping objects like:

```JSON
{
  "siem.__prop1": "some value",
  "siem.__prop2": "some other value"
}
```

Technically speaking, it adheres to the `DYNAMIC_KEY` contract, although I agree that indexing that into ES might end up as:

```JSON
{
  "siem": {
    "__prop1": "some value",
    "__prop2": "some other value"
  }
}
```

FWIW, I don't think it's such an issue:

* If we declare `count_connector_types_by_consumers` as `type: "flattened"`, it won't break. 
* If we declare `dynamic_templates` with `match_path: "count_connector_types_by_consumers.*`, it won't break.
* If we explicitly declare their mappings, it won't break either.

The only potential error would be if we also report the `"siem"` key on its own. Do we?

Reply:
_The only potential error would be if we also report the "siem" key on its own. Do we?_

No, this is not possible.

And thank you for clarifying `DYNAMIC_KEY` usage :)